### PR TITLE
feature/EFRS-589 connect to api

### PR DESCRIPTION
**Pull request by mjakowiec**
PR contains my changes about task EFRS-590

### CODE REVIEW

File: ui/src/environments/environment.dev.ts [hunk 1]
```diff
@@ -20,5 +20,7 @@ export const environment: Environment = {
   production: true,
   basicToken: 'Basic Q29tbW9uQ2xpZW50SWQ6cGFzc3dvcmQ=',
   apiUrl: 'http://compreface-api:8080/admin/',
```
Comment (isobolev) [inline ui/src/environments/environment.dev.ts:22]: is apiUrl is used anywhere? please remove if not

File: ui/src/app/features/drag-n-drop.container/drag-n-drop.container.component.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      [model]="model$ | async">
+   </app-drag-n-drop>`
+})
+export class DragNDropContainerComponent implements OnInit {
+  model$: Observable<Model>;
+
+  constructor(private breadcrumbsFacade: BreadcrumbsFacade) { }
+
+  ngOnInit() {
+    this.model$ = this.breadcrumbsFacade.model$;
```
Comment (isobolev) [inline ui/src/app/features/drag-n-drop.container/drag-n-drop.container.component.ts:34]: should it be modelFacade?

File: ui/src/app/features/drag-n-drop/drag-n-drop.component.html [hunk 3]
```diff
@@ -1,7 +1,35 @@
...
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+  ~ or implied. See the License for the specific language governing
+  ~ permissions and limitations under the License.
+  -->
+
 <h3>{{'dnd.test' | translate}}</h3>
-<div class="dropzone" appDnd (fileDropped)="onFileDropped($event)">
-    <input type="file" #fileDropRef id="fileDropRef" multiple (change)="fileBrowseHandler($event.target.files)" />
+<button mat-button class="mat-button mat-button-base mat-primary ng-star-inserted" color="primary" (click)="trainModel()">
+    {{'dnd.train' | translate}}
```
Comment (isobolev) [inline ui/src/app/features/drag-n-drop/drag-n-drop.component.html:19]: need to make it look like on wireframes

File: ui/src/app/features/drag-n-drop/drag-n-drop.component.html [hunk 4]
```diff
@@ -1,7 +1,35 @@
...
+</button>
+<h3>{{'dnd.add' | translate}}</h3>
+<div class="dropzone" appDnd (fileDropped)="onFileDroppedAdd($event)">
+    <input type="file" #fileDropRef id="fileDropRef" multiple (change)="fileBrowseHandlerAdd($event.target.files)" />
     <img src="../../../assets/dnd/tmp-upload-file.svg" alt="">
     <h3>{{'dnd.title' | translate}}</h3>
     <label>{{'dnd.label' | translate}}</label>
 </div>
+<h3>{{'dnd.recognize' | translate}}</h3>
+<div class="dropzone" appDnd (fileDropped)="onFileDroppedReco($event)">
```
Comment (isobolev) [inline ui/src/app/features/drag-n-drop/drag-n-drop.component.html:29]: why there are 2 dropzones?

File: ui/src/app/features/drag-n-drop/drag-n-drop.component.ts [hunk 5]
```diff
@@ -9,27 +27,79 @@ export class DragNDropComponent {
...
    * @param files (Files List)
    * TODO Send file to api
    */
-  processFile(files: Array<any>) {
-    this.file = files[0];
+  processFileAddToModel(files: Array<any>) {
+    for (const item of files) {
+      this.file = item;
+    }
+    this.dragService.addFace(this.file, this.model).subscribe((res) => {
```
Comment (isobolev) [inline ui/src/app/features/drag-n-drop/drag-n-drop.component.ts:73]: should not call service explicitly. need to use ngrx

File: ui/src/app/features/drag-n-drop/drag-n-drop.component.ts [hunk 6]
```diff
@@ -9,27 +27,79 @@ export class DragNDropComponent {
...
+   * Recognize face
+   * @param files (Files List)
+   * TODO Send file to api
+   */
+  processFileRecoFace(files: Array<any>) {
+    for (const item of files) {
+      this.file = item;
+    }
+    console.log(files.length);
+    this.dragService.recognize(this.file, this.model).subscribe((res) => {
```
Comment (isobolev) [inline ui/src/app/features/drag-n-drop/drag-n-drop.component.ts:89]: same as above

File: ui/src/app/features/drag-n-drop/drag-n-drop.component.ts [hunk 7]
```diff
@@ -9,27 +27,79 @@ export class DragNDropComponent {
...
+  /**
+   * Recognize face
+   * @param files (Files List)
+   * TODO Send file to api
+   */
+  processFileRecoFace(files: Array<any>) {
+    for (const item of files) {
+      this.file = item;
+    }
+    console.log(files.length);
```
Comment (isobolev) [inline ui/src/app/features/drag-n-drop/drag-n-drop.component.ts:88]: remove console log

File: ui/src/app/features/drag-n-drop/drag-n-drop.component.ts (hunk 6 above)
Comment (isobolev) [inline ui/src/app/features/drag-n-drop/drag-n-drop.component.ts:89]: no unsubscribe. memory leak

File: ui/src/app/features/drag-n-drop/drag-n-drop.component.ts [hunk 8]
```diff
@@ -9,27 +27,79 @@ export class DragNDropComponent {
...
+      this.data = res;
+      this.loading = false;
+    });
+  }
+
+  /**
+   * Train model
+   */
+  trainModel() {
+    this.dragService.train(this.model).subscribe((res) => {
```
Comment (isobolev) [inline ui/src/app/features/drag-n-drop/drag-n-drop.component.ts:99]: same as above. The dump component should only emit value through Output. The container component should dispatch the action. API calls should be done in effects

File: ui/src/app/features/drag-n-drop.container/drag-n-drop.container.component.ts [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+    <app-drag-n-drop
+      [model]="model$ | async"
+      [data]="data$ | async"
+      [loading]="loading$ | async"
+      (recognizeFace)=recognizeFace($event)>
+   </app-drag-n-drop>`
+})
+export class DragNDropContainerComponent implements OnInit {
+  model$: Observable<Model>;
+  testSub: Observable<string>;
```
Comment (isobolev) [inline ui/src/app/features/drag-n-drop.container/drag-n-drop.container.component.ts:37]: testSub is not used

File: ui/src/app/features/drag-n-drop.container/drag-n-drop.container.component.ts (hunk 2 above)
Comment (isobolev) [reply ui/src/app/features/drag-n-drop.container/drag-n-drop.container.component.ts:43]: ?

File: ui/src/app/features/drag-n-drop.container/drag-n-drop.container.component.ts [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+  constructor(private breadcrumbsFacade: BreadcrumbsFacade, private dragNdropFacade: DragNDropFacade) {}
+
+  ngOnInit() {
+    this.model$ = this.breadcrumbsFacade.model$;
+    this.testSub = this.dragNdropFacade.testData$;
+  }
+
+  recognizeFace(file: any, model: Model) {
+    this.dragNdropFacade.recognizeFace(model, file);
+
```
Comment (isobolev) [inline ui/src/app/features/drag-n-drop.container/drag-n-drop.container.component.ts:49]: empty line

File: ui/src/app/features/drag-n-drop.container/drag-n-drop.container.module.ts [hunk 11]
```diff
@@ -0,0 +1,31 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+/*
```
Comment (isobolev) [inline ui/src/app/features/drag-n-drop.container/drag-n-drop.container.module.ts:3]: copyright info should start from the first line

File: ui/src/app/features/drag-n-drop/drag-n-drop.component.scss [hunk 12]
```diff
@@ -48,4 +48,14 @@
...
   }
+
+  .mat-button {
+    float: right;
+    font-size: 20px;
+  }
+
+  canvas {
+    width: 100%;
+    height: 1000;
```
Comment (isobolev) [inline ui/src/app/features/drag-n-drop/drag-n-drop.component.scss:59]: add px

File: ui/src/app/features/drag-n-drop/drag-n-drop.component.ts [hunk 13]
```diff
@@ -1,4 +1,23 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import { Component, ViewChild, ElementRef, Input, Output } from '@angular/core';
+import { Model } from 'src/app/data/model';
+import { DragNDropService } from './drag-n-drop.service';
+import { EventEmitter } from 'protractor';
```
Comment (isobolev) [inline ui/src/app/features/drag-n-drop/drag-n-drop.component.ts:20]: incorrect EventEmitter; should be from from '@angular/core'

File: ui/src/app/features/drag-n-drop/drag-n-drop.component.ts [hunk 14]
```diff
@@ -9,27 +28,63 @@ export class DragNDropComponent {
 
   @ViewChild('fileDropRef', { static: false }) fileDropEl: ElementRef;
   file: any;
+  @Input() data: object;
+  @Input() loading = true;
+  @Input() model: Model;
+  @Output() recognizeFace = new EventEmitter();
+  @ViewChild('myCanvas', { static: true }) myCanvas: ElementRef;
+
```
Comment (isobolev) [inline ui/src/app/features/drag-n-drop/drag-n-drop.component.ts:36]: empty line

File: ui/src/app/features/drag-n-drop/drag-n-drop.component.ts [hunk 15]
```diff
@@ -9,27 +28,63 @@ export class DragNDropComponent {
...
+  onFileDroppedReco($event) {
+    this.processFileRecoFace($event);
   }
 
   /**
    * handle file from browsing
    */
-  fileBrowseHandler(files) {
-    this.processFile(files);
+  fileBrowseHandlerReco(files) {
```
Comment (isobolev) [inline ui/src/app/features/drag-n-drop/drag-n-drop.component.ts:50]: please rename without Reco shorthand

File: ui/src/app/features/drag-n-drop/drag-n-drop.component.ts [hunk 16]
```diff
@@ -9,27 +28,63 @@ export class DragNDropComponent {
...
 
   /**
-   * Process file
+   * Recognize face
    * @param files (Files List)
    * TODO Send file to api
    */
-  processFile(files: Array<any>) {
-    this.file = files[0];
+  processFileRecoFace(files: Array<any>) {
```
Comment (isobolev) [inline ui/src/app/features/drag-n-drop/drag-n-drop.component.ts:59]: please rename to processFile

File: ui/src/app/features/drag-n-drop/drag-n-drop.component.ts [hunk 17]
```diff
@@ -9,27 +28,63 @@ export class DragNDropComponent {
...
+    for (const item of files) {
```
Comment (isobolev) [inline ui/src/app/features/drag-n-drop/drag-n-drop.component.ts:60]: redundant. 
change to this.file = files[0];

File: ui/src/app/store/test-model/effects.ts [hunk 18]
```diff
@@ -0,0 +1,45 @@
...
+  ) { }
+
+  @Effect()
+  recognizeFace$ = this.actions.pipe(
+    ofType(recognizeFace),
+    switchMap(action => this.testService.recognize(action.file, action.model).pipe(
+      map(model => recognizeFaceSuccess({ model })),
+      catchError(error => of(recognizeFaceFail({ error }))),
+    )),
+  );
```
Comment (isobolev) [inline ui/src/app/store/test-model/effects.ts:43]: please add effect to listen for recognizeFaceFail and display an error message if any

File: ui/src/app/features/drag-n-drop.container/drag-n-drop.container.component.ts [hunk 19]
```diff
@@ -0,0 +1,51 @@
...
+import { DragNDropFacade } from './drag-n-drop.facade';
+import { async } from '@angular/core/testing';
+
+@Component({
+  selector: 'app-drag-n-drop-container',
+  template: `
+    <app-drag-n-drop
+      [model]="model$ | async"
+      [data]="data$ | async"
+      [loading]="loading$ | async"
```
Comment (isobolev) [inline ui/src/app/features/drag-n-drop.container/drag-n-drop.container.component.ts:31]: loading$ and data$ are not defined in the component. in reducer loading prop is called isPending

File: ui/src/app/store/test-model/reducers.ts [hunk 20]
```diff
@@ -0,0 +1,48 @@
...
+
+const initialState: TestEntityState = testAdapter.getInitialState({
+  isPending: false
+});
+
+const reducer: ActionReducer<TestEntityState> = createReducer(
+  initialState,
+  on(recognizeFace, (state) => ({ ...state, isPending: true })),
+  on(recognizeFaceSuccess, (state,  { model } ) => {
+    return model;
```
Comment (isobolev) [inline ui/src/app/store/test-model/reducers.ts:41]: should extend the current state. just `return model` will override `isPending` property

File: ui/src/environments/environment.dev.ts (hunk 1 above)
Comment (isobolev) [reply ui/src/environments/environment.dev.ts:22]: ?

File: ui/src/app/features/drag-n-drop.container/drag-n-drop.container.component.ts [hunk 21]
```diff
@@ -0,0 +1,48 @@
...
+import { AppState } from 'src/app/store';
+import { DragNDropFacade } from './drag-n-drop.facade';
+import { async } from '@angular/core/testing';
+
+@Component({
+  selector: 'app-drag-n-drop-container',
+  template: `
+    <app-drag-n-drop
+      [model]="testMmodel$ | async"
+      [data]="model$ | async"
```
Comment (isobolev) [inline ui/src/app/features/drag-n-drop.container/drag-n-drop.container.component.ts:30]: model$ and isPending$ is not defined in component

File: ui/src/app/features/drag-n-drop/drag-n-drop.component.ts (hunk 13 above)
Comment (isobolev) [reply ui/src/app/features/drag-n-drop/drag-n-drop.component.ts:20]: not resolved

File: ui/src/app/features/drag-n-drop/drag-n-drop.component.ts (hunk 15 above)
Comment (isobolev) [reply ui/src/app/features/drag-n-drop/drag-n-drop.component.ts:50]: not resolved

File: ui/src/app/features/drag-n-drop.container/drag-n-drop.container.component.ts [hunk 22]
```diff
@@ -0,0 +1,48 @@
...
+import { DragNDropFacade } from './drag-n-drop.facade';
+import { async } from '@angular/core/testing';
+
+@Component({
+  selector: 'app-drag-n-drop-container',
+  template: `
+    <app-drag-n-drop
+      [model]="model$ | async"
+      [data]="data$ | async"
+      (recognizeFace)=recognizeFace($event)>
```
Comment (isobolev) [inline ui/src/app/features/drag-n-drop.container/drag-n-drop.container.component.ts:30]: need to pass isPending or loading property

File: ui/src/app/features/drag-n-drop.container/drag-n-drop.container.component.ts [hunk 23]
```diff
@@ -0,0 +1,48 @@
...
+   </app-drag-n-drop>`
+})
+export class DragNDropContainerComponent implements OnInit {
+  model$: Observable<Model>;
+  private store: Store<AppState>;
```
Comment (isobolev) [inline ui/src/app/features/drag-n-drop.container/drag-n-drop.container.component.ts:35]: unused declaration

File: ui/src/app/features/drag-n-drop/drag-n-drop.component.ts [hunk 24]
```diff
@@ -9,27 +28,60 @@ export class DragNDropComponent {
 
   @ViewChild('fileDropRef', { static: false }) fileDropEl: ElementRef;
   file: any;
+  @Input() data: object;
+  @Input() loading = true;
```
Comment (isobolev) [inline ui/src/app/features/drag-n-drop/drag-n-drop.component.ts:32]: loading is always true

File: ui/src/app/features/drag-n-drop/drag-n-drop.component.ts [hunk 25]
```diff
@@ -1,35 +1,52 @@
...
+  @Output() upload: EventEmitter<{file: File, model: Model}> = new EventEmitter();
+  private file: any;
+
+  constructor(private translate: TranslateService) {
+    // Set the default title and label. But leave possibility to set another title and label.
+    this.title = this.translate.instant('dnd.title');
+    this.label = this.translate.instant('dnd.label');
+  }
+
+  ngOnInit() {
```
Comment (isobolev) [inline ui/src/app/features/drag-n-drop/drag-n-drop.component.ts:25]: please remove empty onInit

File: ui/src/app/features/drag-n-drop/drag-n-drop.component.ts [hunk 26]
```diff
@@ -1,35 +1,52 @@
...
   }
 
   /**
-   * Process file
+   * Recognize face
    * @param files (Files List)
    * TODO Send file to api
    */
-  processFile(files: Array<any>) {
+  UploadFile(files: Array<any>) {
```
Comment (isobolev) [inline ui/src/app/features/drag-n-drop/drag-n-drop.component.ts:48]: method name should start from the lowercase character

File: ui/src/app/features/face-recognition/face-recognition.component.ts [hunk 27]
```diff
@@ -0,0 +1,39 @@
...
+import { AppState } from '../../store';
+import { recognizeFace, addFace } from '../../store/face-recognition/actions';
+
+@Component({
+  selector: 'app-face-recognition',
+  templateUrl: './face-recognition.component.html',
+  styleUrls: ['./face-recognition.component.scss']
+})
+export class FaceRecognitionComponent implements OnInit {
+  public model$: Observable<Model>;
```
Comment (isobolev) [inline ui/src/app/features/face-recognition/face-recognition.component.ts:14]: public is redundant

**Comment by olehkhomyk** [comment]
Close this pr because old changes

**Pull request closed without merging**